ground/src/main/java/com/google/android/ground/ui/datacollection/tasks/multiplechoice/SelectOneOptionAdapter.kt

```diff
@@ -47,24 +47,26 @@ class SelectOneOptionAdapter(
     holder.bind(options[position])
 
     holder.binding.radioButton.isChecked = position == selectedIndex
-
     holder.binding.radioButton.setOnClickListener {
-      val oldPosition = selectedIndex
-      selectedIndex = holder.adapterPosition
-      viewModel.updateResponse(listOf(options[selectedIndex]))
+      handleItemStateChange(it, holder.adapterPosition)
+    }
+  }
+
+  private fun handleItemStateChange(view: View, position: Int) {
+    val oldPosition = selectedIndex
+    selectedIndex = position
+    handleOptionSelected.invoke(listOf(options[selectedIndex]))
```

Comment:
Should be able to just call it like a function `handleOptionSelected(listOf(options[selectedIndex]))`